### PR TITLE
feat(call-analyzer): fix invalid client.call.leave leaveReasons

### PR DESCRIPTION
**Pull request by chburket**
## COMPLETES

https://jira-eng-gpk2.cisco.com/jira/browse/SPARK-458903

## This pull request addresses

Currently we are using one of `MEETING_REMOVED_REASON` as the CA event leaveReason.  These are not valid values for the CA event, which needs to be one of 'paired-leave' | 'one-to-one' | 'ended-by-locus'.

We should not conflate locus leave reason with CA event leave reason.

## by making the following changes

I added a new prop/arg to meeting.leave() (`clientEventLeaveReason`) that can be used to pass in one of the valid CA reasons. By default this property will be undefined (it is optional).

### Change Type

- [X] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation update
- [ ] Tooling change
- [ ] Internal code refactor

## The following scenarios where tested

Updated unit tests and tested manually.

### I certified that

- [X] I have read and followed [contributing guidelines](https://github.com/webex/webex-js-sdk/blob/master/CONTRIBUTING.md#submitting-a-pull-request)
- [X] I discussed changes with code owners prior to submitting this pull request

- [X] I have not skipped any automated checks
- [X] All existing and new tests passed
- [ ] I have updated the documentation accordingly

### CODE REVIEW

**Review by arun3528 — APPROVED** [review]

File: packages/@webex/plugin-meetings/src/meeting/index.ts [hunk 1]
```diff
@@ -5910,8 +5914,15 @@ export default class Meeting extends StatelessWebexPlugin {
    * @public
    * @memberof Meeting
    */
-  public leave(options: {resourceId?: string; reason?: any} = {} as any) {
+  public leave(
+    options: {
+      resourceId?: string;
+      reason?: any;
```
Comment (marcin-bazyl) [inline packages/@webex/plugin-meetings/src/meeting/index.ts:5920]: Looks like `reason` is now not used for anything? its value is copied into `leaveReason`, but then `leaveReason` is not passed anywhere further, only logged in 1 place: in line 5978 and we should probably log `options.clientEventLeaveReason` there instead, right? 
Or am I missing something?

File: packages/@webex/plugin-meetings/src/meeting/index.ts [hunk 2]
```diff
@@ -5910,8 +5914,15 @@ export default class Meeting extends StatelessWebexPlugin {
...
+      clientEventLeaveReason?: ClientEventLeaveReason;
```
Comment (marcin-bazyl) [inline packages/@webex/plugin-meetings/src/meeting/index.ts:5921]: how are client apps supposed to know the difference between `reason` and `clientEventLeaveReason` and which one they should set? Do we really need 2 of them? Also jsdoc needs updating

**Review by marcin-bazyl — CHANGES_REQUESTED** [review]

**Review by marcin-bazyl — APPROVED** [review]

File: packages/@webex/plugin-meetings/src/meeting/index.ts (hunk 1 above)
Comment (chburket) [reply packages/@webex/plugin-meetings/src/meeting/index.ts]: The "reason" arg is now only used for logging. Several places that call meeting.leave() pass in a specific reason, and two other places use the reason from locus events (EVENTS.DESTROY_MEETING, EVENTS.REMOTE_RESPONSE).

The "clientEventLeaveReason" is used for the CA event leaveReason. It must be one of three values. With this change, the "clientEventLeaveReason" will not be anything other than undefined (which is perfectly fine, it's optional), but I will open another Jira to see if we should make use of this arg from some of our meeting.leave() calls

File: packages/@webex/plugin-meetings/src/meeting/index.ts (hunk 2 above)
Comment (chburket) [reply packages/@webex/plugin-meetings/src/meeting/index.ts:5924]: Updated js doc

**Comment by chburket** [comment]
opening a new branch

**Pull request closed without merging**